src/async_impl/client.rs

```diff
@@ -1858,24 +1858,14 @@ impl ClientBuilder {
 
     /// Override DNS resolution for specific domains to a particular IP address.
     ///
-    /// Warning
-    ///
-    /// Since the DNS protocol has no notion of ports, if you wish to send
-    /// traffic to a particular port you must include this port in the URL
-    /// itself, any port in the overridden addr will be ignored and traffic sent
-    /// to the conventional port for the given scheme (e.g. 80 for http).
+    /// Set the port to `0` to use the conventional port for the given scheme (e.g. 80 for http).
```

Comment:
It is my understanding the that URL can also override the default port from the scheme.  Perhaps "use the port specified in the URL or default for the URL's scheme" or something might be clearer?